functions-and-operators/string-functions.md

```diff
@@ -179,11 +179,198 @@ Return the length of a string in bytes.
+
+To specify the different escape character, you can use `ESCAPE` clause.
+```sql
+SELECT 'sushi_' LIKE 'sushi*_' ESCAPE '*' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+
+SELECT 'sushi!' LIKE 'sushi*_' ESCAPE '*' AS result;
++--------+
+| result |
++--------+
+|      0 |
++--------+
+```
+
+`LIKE` permits numeric expressions.
+```sql
+SELECT 10 LIKE '1%' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+
+SELECT 10000 LIKE '12%' AS result;
++--------+
+| result |
++--------+
+|      0 |
++--------+
+```
 
 ### [`LOCATE()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_locate)
 
-Return the position of the first occurrence of substring.
+`LOCATE(substr, str[, pos])` returns the position of first occurrence of substring in string.
+When you specify `pos`, `LOCATE()` starts seeking at the position `pos`.
+`LOCATE` returns `0` if `substr` not in `str` and returns `NULL` if any arguments are null.
+This function is multi-byte safe and is case-sensitive only if at least one argument is a binary string.
+```sql
+SELECT LOCATE('bar', 'foobarbar');
++----------------------------+
+| LOCATE('bar', 'foobarbar') |
++----------------------------+
+|                          4 |
++----------------------------+
+
+SELECT LOCATE('baz', 'foobarbar');
++----------------------------+
+| LOCATE('baz', 'foobarbar') |
++----------------------------+
+|                          0 |
++----------------------------+
+
+SELECT LOCATE('bar', 'fooBARBAR');
++----------------------------+
+| LOCATE('bar', 'fooBARBAR') |
++----------------------------+
+|                          4 |
++----------------------------+
+
+SELECT LOCATE('bar', 'fooBARBAR', 100);
++---------------------------------+
+| LOCATE('bar', 'fooBARBAR', 100) |
++---------------------------------+
+|                               0 |
++---------------------------------+
+
+SELECT LOCATE('bar', 'fooBARBAR', 5);
++-------------------------------+
+| LOCATE('bar', 'fooBARBAR', 5) |
++-------------------------------+
+|                             7 |
++-------------------------------+
+
+SELECT LOCATE('bar', NULL);
++---------------------+
+| LOCATE('bar', NULL) |
++---------------------+
+|                NULL |
++---------------------+
+
+SELECT LOCATE('い', 'たいでぃーびー');
++----------------------------------------+
+| LOCATE('い', 'たいでぃーびー')         |
++----------------------------------------+
+|                                      2 |
++----------------------------------------+
+
+SELECT LOCATE('い', 'たいでぃーびー', 3);
++-------------------------------------------+
+| LOCATE('い', 'たいでぃーびー', 3)         |
++-------------------------------------------+
+|                                         0 |
++-------------------------------------------+
+
+SELECT LOCATE('🍺', '🍣🍣🍣🍺🍺');
++----------------------------------------+
+| LOCATE('🍺', '🍣🍣🍣🍺🍺')            |
++----------------------------------------+
+|                                      1 |
```

Comment:
I'm getting 4 and not 1

```
mysql> SELECT LOCATE('🍺', '🍣🍣🍣🍺🍺');
+----------------------------------------+
| LOCATE('🍺', '🍣🍣🍣🍺🍺')                         |
+----------------------------------------+
|                                      4 |
+----------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT TIDB_VERSION()\G
*************************** 1. row ***************************
TIDB_VERSION(): Release Version: v7.5.0
Edition: Community
Git Commit Hash: 069631e2ecfedc000ffb92c67207bea81380f020
Git Branch: heads/refs/tags/v7.5.0
UTC Build Time: 2023-11-24 08:50:14
GoVersion: go1.21.3
Race Enabled: false
Check Table Before Drop: false
Store: tikv
1 row in set (0.00 sec)
```

Reply:
same with https://github.com/pingcap/docs/pull/16084#discussion_r1448222749

Reply:
What TiDB version and MySQL client are you testing with?